continuous-testing-foundations/dotnet/Pragmatest.Wallets/test/Pragmatest.Wallets.Web.UnitTests/WalletControllerUnitTests.cs

```diff
@@ -39,14 +42,134 @@ public async Task Balance_GetBalanceIs4_ReturnsOkAndBalance4()
             // Act 
             ActionResult<BalanceResponse> actionResult = await walletController.Balance();
             ActionResult actualActionResult = actionResult.Result;
-            
+
             // Assert
             OkObjectResult okObjectResult = Assert.IsType<OkObjectResult>(actionResult.Result);
             BalanceResponse actualBalanceResponse = Assert.IsType<BalanceResponse>(okObjectResult.Value);
+
             Assert.Equal(expectedBalanceResponse, actualBalanceResponse);
 
             walletServiceMock.Verify(walletService => walletService.GetBalanceAsync(), Times.Once);
             walletServiceMock.VerifyNoOtherCalls();
         }
+
+        [Fact]
+        public async Task Deposit_Deposit10_ReturnsOkAndBalance15()
+        {
+            // Arrange
+            decimal expectedBalanceAmount = 15;
+            decimal depositAmount = 10;
+
+
+            Balance expectedBalance = new Balance() { Amount = expectedBalanceAmount };
+            BalanceResponse expectedBalanceResponse = new BalanceResponse() { Amount = expectedBalanceAmount };
+
+            ILogger<WalletController> logger = Mock.Of<ILogger<WalletController>>();
+
+            Mock<IWalletService> walletServiceMock = new Mock<IWalletService>();
+
+            Deposit deposit = new Deposit { Amount = depositAmount };
+
+            walletServiceMock
+                .Setup(walletService => walletService.DepositFundsAsync(deposit))
+                .Returns(Task.FromResult(expectedBalance));
+
+            IWalletService walletService = walletServiceMock.Object;
+
+            WithdrawalRequest depositRequest = new WithdrawalRequest { Amount = depositAmount };
```

Comment:
I think this is meant to be a DepositRequest object

Reply:
Fixed